lib/rest.js

```diff
@@ -234,96 +193,43 @@ class BinanceRest {
             query = {
                 timestamp: new Date().getTime()
             }
-        } else if (!query.timestamp) {
-            query.timestamp = new Date().getTime();
+        } else {
+          query = createQueryObject(query,true,true)
         }
-
         return this._makeRequest(query, callback, 'api/v3/account', 'SIGNED');
     }
 
     myTrades(query = {}, callback) {
-        if (_.isString(query)) {
-            query = {
-                symbol: query
-            }
-        }
-        if (!query.timestamp) {
-            query.timestamp = new Date().getTime();
-        }
-
-        return this._makeRequest(query, callback, 'api/v3/myTrades', 'SIGNED');
+        return this._makeRequest(createQueryObject(query,true), callback, 'api/v3/myTrades', 'SIGNED');
     }
 
     withdraw(query = {}, callback) {
-        if (!query.timestamp) {
-            query.timestamp = new Date().getTime();
-        }
-
-        return this._makeRequest(query, callback, 'wapi/v3/withdraw.html', 'SIGNED', 'POST');
+        return this._makeRequest(createQueryObject(query,true,true), callback, 'wapi/v3/withdraw.html', 'SIGNED', 'POST');
     }
 
     depositHistory(query = {}, callback) {
-        if (_.isString(query)) {
-            query = {
-                asset: query
-            }
-        }
-        if (!query.timestamp) {
-            query.timestamp = new Date().getTime();
-        }
-
-        return this._makeRequest(query, callback, 'wapi/v3/depositHistory.html', 'SIGNED', 'GET');
+        return this._makeRequest(createQueryObject(query,true), callback, 'wapi/v3/depositHistory.html', 'SIGNED', 'GET');

```

Comment:
This is going to break calls to the function that only pass the string.  This, withdrawHistory, and depositAddress all set `query.asset` and not `query.symbol`.

Also, ideally all the tests would be updated to utilize all cases where a string can be passed so they'd catch this type of error in the future.  It would require setting up sinon's faketimers so the timestamp would remain consistent and assertable.  If you wanna just fix these 3 cases though, I can take care of the tests.

Reply:
Cool, ya np.  I'll update it this weekend.